searchlib/src/vespa/searchlib/attribute/document_weight_or_filter_search.cpp

```diff
@@ -32,7 +32,7 @@ class DocumentWeightOrFilterSearchImpl : public DocumentWeightOrFilterSearch
     }
 
     std::unique_ptr<BitVector> get_hits(uint32_t begin_id) override {
-        return _children.get_hits(begin_id, getEndId());
+        return _children.get_hits(std::min(getEndId(), std::max(begin_id, getDocId())), getEndId());
```

Comment:
this gives me bad memories

Reply:
It failed a conformance test without. I will check for alternative options.

Reply:
Fixed